website/static/js/pages/dashboard-page.js

```diff
@@ -40,6 +41,108 @@ request.fail(function(xhr, textStatus, error) {
+             'If you change your mind, visit the \<a href="/settings/account/">user settings page</a>.';
+        }
+
+        var failSuccessMessage = 'There are a problem adding \<b>' + email[0].address +
+            '\</b>. Please contact <a href="mailto: support@osf.io">support@osf.io</a> if the problem persists.';
+
+        var failCancelMessage = 'There are a problem removing \<b>' + email[0].address +
+            '\</b>. Please contact <a href="mailto: support@osf.io">support@osf.io</a> if the problem persists.';
+
+        bootbox.dialog({
+            title: title,
+            message: mergeMessage,
+            onEscape: function() {},
+            backdrop: true,
+            closeButton: true,
+            buttons: {
+                confirm: {
+                    label: 'Add email',
+                    className: 'btn-success',
+                    callback: function () {
+                        $osf.putJSON(
+                            confirmedEmailURL,
+                            email[0]
+                        ).done(function () {
+                            $osf.growl('Success', confirmMessage, 'success', 3000);
+                            confirm_emails(emails);
+                        }).fail(function (xhr, textStatus, error) {
+                            Raven.captureMessage('Could not add email', {
+                                url: url,
+                                textStatus: textStatus,
+                                error: error
+                            });
+                            $osf.growl('Error',
+                                failSuccessMessage,
+                                'danger'
+                            );
+                        });
+                        confirm_emails(emails);
+                    }
+                },
+                cancel: {
+                    label: 'Cancel',
+                    className: 'btn-default',
+                    callback: function () {
+                        $osf.putJSON(
+                            removeConfirmedEmailURL,
+                            email[0]
+                        ).done(function () {
+                            $osf.growl('Warning', nopeMessage, 'warning', 8000);
+                            confirm_emails(emails);
+                        }).fail(function (xhr, textStatus, error) {
+                            Raven.captureMessage('Could not remove email', {
+                                url: url,
+                                textStatus: textStatus,
+                                error: error
+                            });
+                            $osf.growl('Error',
+                                failCancelMessage,
+                                'danger'
+                            );
+                        });
+                    }
+                }
+            }
+        });
+
+    }
+}
+
+$.getJSON(confirmedEmailURL).done(function(response) {
```

Comment:
This looks like it would send this request out every time this page is loaded, though we only want that to happen on login.

Also, given a) you have merge conflicts, b) this is the `dashboard-page`, and c) a huge rewrite to dashboard/projectOrganizer just went out, this will probably need refactor.